www/content/essays/no-build-step.md

```diff
@@ -0,0 +1,68 @@
+
+Of course, most people's experience with JavaScript is that is ages like than milk. Reopen a node repository after 3 months and you'll find that your project is mired in a flurry of security warnings, backwards-incompatible library "upgrades," and a frontend framework whose cultural peak was the exact moment you started the project and is now widely considered tech debt. Who's to blame for this situation is for someone else to decide, but, in any case, this entire problem class is eliminated if you depend on nothing but the JavaScript runtime.
+
+A popular way to write JavaScript today is compile it from TypeScript (which I will use frequently as an example, because TypeScript is probably the [best reason](https://en.wikipedia.org/wiki/Straw_man#Steelmanning) to use a build system). TypeScript does not run natively in web browsers, so TypeScript code is not protected by [ECMA's](https://developer.mozilla.org/en-US/docs/Glossary/ECMA) fanatical devotion to backwards compatibility. Like any dependency, new major TypeScript versions are not guaranteed to be backwards compatible with the previous ones. They might be! But if they aren't, then you need to do maintenance if you want to use the modern development toolchain.
+
+Maintenance is a cost paid for with labor, and open-source codebases are the projects that can least afford to pay it. Opting not to use a build step drastically minimizes the labor required to keep htmx up-to-date. This experience has been borne out by [intercooler.js](https://intercoolerjs.org), the predecessor to htmx which is maintained indefinitely with (as I understand) very little effort. When htmx 1.0 was released, TypeScript was at version 4.1; when intercooler.js was released, TypeScript was pre-1.0. Would code written in those TypeScript versions compile unmodified in today's TypeScript compiler (version 5.1 at the time of writing)? Maybe, maybe not.
+
+But htmx is written in JavaScript, with no dependencies, so it will run unmodified for as long as web browsers remain relevant. Let the browser vendors do the hard work for you.
+
+## Developer Experience
+It is true that the TypeScript developer experience (DX) is better than the JavaScript developer experience in many respects. TypeScript has... types, which make your IDE better at suggesting code and pointing out where you might have used methods incorrectly. The tools for automatically renaming and refactoring code are much more reliable for TypeScript than they are for JavaScript.
+
```

Comment:
The first sentence does not introduce well to the coming pro/con discussion.

Reply:
It's not really a pro/con discussion. I'm saying "most people take it as a given that TypeScript has better DX, here's a bunch of DX improvements that we allow for by forgoing it." The cons come at the end.

Reply:
You know, thought about this a tad, and I think you're right and I can move a bit of this opening paragraph to the costs section and it will read more smoothly.